tfhe/src/core_crypto/commons/numeric/unsigned.rs

```diff
@@ -28,8 +28,12 @@ pub trait UnsignedInteger:
     + Not<Output = Self>
     + Shl<usize, Output = Self>
     + ShlAssign<usize>
+    + Shl<u32, Output = Self>
+    + ShlAssign<u32>
     + Shr<usize, Output = Self>
     + ShrAssign<usize>
+    + Shr<u32, Output = Self>
+    + ShrAssign<u32>
```

Comment:
looks to create annoying situations, any reason for the u32 additional requirement ?